meson.build

```diff
@@ -514,73 +507,52 @@ if not posix and not win32_desktop
 endif
 
 glob_posix = cc.has_function('glob', prefix: '#include <glob.h>')
-if glob_posix
-    features += 'glob_posix'
-endif
+features += {'glob-posix': glob_posix}
 
 glob_win32 = win32 and not posix
 if glob_win32
-    features += 'glob_win32'
     sources += files('osdep/glob-win.c')
 endif
+features += {'glob-win32': glob_win32}
 
 glob = glob_posix or glob_win32
-if glob
-    features += 'glob'
-endif
+features += {'glob': glob}
 
 vt_h = cc.has_header_symbol('sys/vt.h', 'VT_GETMODE')
-if vt_h
-    features += 'vt.h'
-endif
+features += {'vt.h': vt_h}
 
 consio_h = not vt_h and cc.has_header_symbol('sys/consio.h', 'VT_GETMODE')
-if consio_h
-    features += 'consio.h'
-endif
+features += {'consio.h': consio_h}
 
-fragments = join_paths(source_root, 'waftools', 'fragments')
-
-glibc_thread_name = cc.compiles(files(join_paths(fragments, 'glibc_thread_name.c')),
-                                name: 'glibc-thread-name check')
-if glibc_thread_name and posix
-    features += 'glibc-thread-name'
-endif
+glibc_thread_name = cc.has_function('pthread_setname_np',
+                                    prefix: '#define _GNU_SOURCE\n#include <pthread.h>')
+features += {'glibc-thread-name': glibc_thread_name and posix and not darwin}
```

Comment:
this should keep the old check because it's also a matter of argument order/count and not just function presence
(notice how you had to add `not darwin`)

Reply:
Yeah maybe this one is just doomed.